File: 125-valid-palindrome/125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        // convert to lowercase
        transform(s.begin(), s.end(), s.begin(), ::tolower);
        
        int length = s.length();
        int l = 0;
        int r = length - 1;
    
        // loop through string, use two pointer
        for (int i = 0; i < length; i++) {
            // if s[l] non alphanum, go to next char
            if (!isalnum(s[l])) l++;
            
            // if s[r] non alphanum, go to next char
            else if (!isalnum(s[r])) r--;
            
            // if both chars are good, compare if same. if so, move both pointers
            else if (s[l] == s[r]) {
                l++;
                r--;
            }
            
            // if different, not palindrome
            else return false;
        }
        
        return true;
    }
};

File: 125-valid-palindrome/125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        s = s.lower()
        l, r = 0, len(s)-1
        if len(s) == 1: return True
        while l <= r:
            while l<=r and not s[l].isalnum(): l+=1
                
            while l<=r and not s[r].isalnum(): r-=1
                
            if l < r and s[l] != s[r]: return False
            
            l+=1
            r-=1
            
        return True